pkg/riff/commands/credential_list_test.go

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package commands_test
+
+import (
+	"testing"
+
+	"github.com/projectriff/riff/pkg/riff/commands"
+	rifftesting "github.com/projectriff/riff/pkg/testing"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestCredentialListCommand(t *testing.T) {
+	rifftesting.Table{{
```

Comment:
it's subjective but I find this particular structure of tests to be be a bit hard to read (especially with the actual invocation is being at the bottom of the file which breaks the reading linearity)
also: how do you handle error cases with that structure?

Reply:
I'm wondering if that approach does not end up sharing too much. Can all the test cases fit in this structure? Having an error case in the PR would help.

Reply:
With this model of testing we can still create one-off go test cases. There is nothing forcing all tests to use the table model. With Ginkgo, it's odd if some of your tests are not managed by ginkgo.

As for error cases, the fake clients have reactors that are able to receive and mutate all requests and also return custom errors. I'll work up a test case with a network failure as an example.

Reply:
@fbiville check out https://github.com/projectriff/riff/blob/84bce9fd5ded95cb72ad85e3b6df0157f90a0621/pkg/riff/commands/credential_list_test.go#L98-L103